controllers/runner_controller.go

```diff
@@ -274,6 +279,12 @@ func (r *RunnerReconciler) processRunnerDeletion(runner v1alpha1.Runner, ctx con
 		log.Info("Removed finalizer")
 	}
 
+	if runner.Spec.ContainerMode == "kubernetes" {
+		if err := r.deletePersistentVolume(ctx, &runner); err != nil {
```

Comment:
Create PV only if storage class is `local-storage`, and delete only if storage class is `local-storage`

Reply:
We spoke in person but wanted to confirm, we shouldn't have to create anything. For stateful sets we will set this as VolumeClaimTemplate

For runners we will set an [ephemeral volume](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volumes)

Most cloud k8s providers will create pv/pvc automatically, but we may need to create some locally for testing

Reply:
This also means we don't need to clean up volumes!